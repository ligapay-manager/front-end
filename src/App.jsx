import React, { Component } from 'react';
import { Animated, Easing, View } from 'react-native';

import styled from 'styled-components/native';
import Button from './components/Button';


const Container = styled.View`
  background-color: #178c58;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
`;
const Title = Animated.createAnimatedComponent(styled.Text`
  color: #c6c013;
  font-size: 60px;
  font-family: 'Pacifico-Regular';
  margin-bottom: 40px;
  padding: 0px 10px;
`);

const Input = Animated.createAnimatedComponent(styled.TextInput`
  background-color: #fff;
  height: 38px;
  border-radius: 2px;
  elevation: 2px;
  padding: 0px 10px;
  width: 210px;
  margin-bottom: 10px;
  text-align: center;
`);

const AnimatedButton = Animated.createAnimatedComponent(Button);

export default class App extends Component {
  constructor(props) {
    super(props);

    this.animations = {
      titlePosition: new Animated.Value(-40),
      usernamePosition: new Animated.Value(-40),
      passwordPosition: new Animated.Value(40),
      buttonPosition: new Animated.Value(40),
      fade: new Animated.Value(0),
    };
  }

  componentDidMount() {
    Animated.parallel([
      Animated.timing(this.animations.titlePosition, {
        toValue: 0,
        duration: 800,
        easing: Easing.ease,
        useNativeDriver: true
      }),
      Animated.timing(this.animations.fade, {
        toValue: 1,
        duration: 800,
        easing: Easing.ease,
        useNativeDriver: true
      }),
      Animated.timing(this.animations.usernamePosition, {
        toValue: 0,
        duration: 800,
        easing: Easing.ease,
        useNativeDriver: true
      }),
      Animated.timing(this.animations.passwordPosition, {
        toValue: 0,
        duration: 800,
        easing: Easing.ease,
        useNativeDriver: true
      }),
      Animated.timing(this.animations.buttonPosition, {
        toValue: 0,
        duration: 800,
        easing: Easing.ease,
        useNativeDriver: true
      })

    ]).start();
  }

  render() {
    const {
      fade,
      titlePosition,
      buttonPosition,
      passwordPosition,
      usernamePosition
    } = this.animations;

    return (
      <Container>

        <Title style={{ opacity: fade, transform: [{ translateY: titlePosition }] }}>
          LigaPay
        </Title>

        <Input
          style={{ opacity: fade, transform: [{ translateX: usernamePosition }] }}
          placeholder="UsuÃ¡rio"
        />

        <Input
          style={{ opacity: fade, transform: [{ translateX: passwordPosition }] }}
          placeholder="Senha"
          secureTextEntry
        />

        <AnimatedButton
          style={{ opacity: fade, transform: [{ translateY: buttonPosition }] }}
          title="Login"
        />
      </Container>
    );
  }
}
