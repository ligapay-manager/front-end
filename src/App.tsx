import React, { Component } from 'react';
import { Animated, Easing } from 'react-native';

import styled from "styled-components/native";
import Button from "./components/Button";

const Container = styled.View`
  background-color: #178c58;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
`
const Title = Animated.createAnimatedComponent(styled.Text`
  color: #c6c013;
  font-size: 60px;
  font-family: 'Pacifico-Regular';
  margin-bottom: 40px;
  padding: 0px 10px;
`)

const Input = styled.TextInput`
  background-color: #fff;
  height: 38px;
  border-radius: 2px;
  elevation: 2px;
  padding: 0px 10px;
  width: 210px;
  margin-bottom: 10px;
  text-align: center;
`

export default class App extends Component<Props> {
  titlePosition: Animated.Value;
  titleFade: Animated.Value;

  constructor(props: Props) {
    super(props)

    this.titlePosition = new Animated.Value(-40)
    this.titleFade = new Animated.Value(0)
  }

  componentDidMount() {
    Animated.parallel([
      Animated.timing(this.titlePosition, {
        toValue: 0,
        duration: 800,
        easing: Easing.ease,
        useNativeDriver: true
      }),
      Animated.timing(this.titleFade, {
        toValue: 1,
        duration: 800,
        easing: Easing.ease,
        useNativeDriver: true
      })
    ]).start()
  }

  render() {
    const { titleFade, titlePosition } = this

    return (
      <Container>
        <Title style={{ opacity: titleFade, transform: [{ translateY: titlePosition }] }}>
          LigaPay
        </Title>

        <Input placeholder={'UsuÃ¡rio'} />
        <Input placeholder={'Senha'} secureTextEntry />
        <Button title={'Login'} />
      </Container >
    );
  }
}

interface Props { }
