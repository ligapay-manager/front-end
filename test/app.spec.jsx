// test/app-spec-jsx
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App';

// needed for animations
jest.useFakeTimers();


it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});
