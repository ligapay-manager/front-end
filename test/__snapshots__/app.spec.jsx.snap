// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#178c58",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#c6c013",
          "fontFamily": "Pacifico-Regular",
          "fontSize": 60,
          "marginBottom": 40,
          "paddingBottom": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 0,
        },
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateY": -40,
            },
          ],
        },
      ]
    }
  >
    LigaPay
  </Text>
  <TextInput
    allowFontScaling={true}
    placeholder="UsuÃ¡rio"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "elevation": 2,
          "height": 38,
          "marginBottom": 10,
          "paddingBottom": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 0,
          "textAlign": "center",
          "width": 210,
        },
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateX": -40,
            },
          ],
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    placeholder="Senha"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "elevation": 2,
          "height": 38,
          "marginBottom": 10,
          "paddingBottom": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 0,
          "textAlign": "center",
          "width": 210,
        },
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateX": 40,
            },
          ],
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;
